Homework Done

goldovskaya-m opened the pull request:
Проверьте ДЗ

goldovskaya-m committed:
Homework Done

goldovskaya-m committed:
Внесла изменения после проверки

AleksandrZhukovJava commented on src/Main.java:
```diff
@@ -1,5 +1,78 @@
         System.out.println("Homework9");
+        System.out.println("Задача 1");
+        double[] monday = {5, 6, 7, 8, 9};
+        double SUM = 0;
```
большими буквами мы делаем только переменные константы (вы их пока не проходили), сейчас пока все маленькими

AleksandrZhukovJava commented on src/Main.java:
```diff
@@ -1,5 +1,78 @@
+        double maxMonday2 = monday[0];
+        for (int i = 0; i < monday.length; i++) {
+            if (monday[i] > minMonday2) {
+                maxMonday2 = monday[i];
```
а где minMonday вычисляется =)

AleksandrZhukovJava commented on src/Main.java:
```diff
@@ -1,5 +1,78 @@
+            SUM2 += monday[i];
+        }
+        double averageWeek = ((double) SUM2 / monday.length) ;
+        System.out.printf("Средняя сумма трат за месяц  35/5 = " +
```
лучше не хардкодить числа в выводе, тогда придется их менять если вдруг в массив добавиться число. Если хочется всё таки их тут иметь, сделай динамическими, что бы сами менялись в зависимости от условий. То есть используй переменные

AleksandrZhukovJava commented on src/Main.java:
```diff
@@ -1,5 +1,78 @@
+        System.out.println("Задача 4");
+        char[] reverseFullName = {'n', 'a', 'v', 'I', ' ', 'v', 'o', 'n', 'a', 'v', 'I'};
+        char temp;
+        for (int i = 0; i < reverseFullName.length && i < 5; i++) {
```
А если фио будет побольше? Всё сломается из за второго условия. В книге сотрудника не обязательно же только сотрудники из 10-11 букв. Любую программу старайся конструировать так, что бы она работала с любым набором данных

AleksandrZhukovJava commented on src/Main.java:
```diff
@@ -1,5 +1,78 @@
+        // 2 вариант Задача 4
+        char[] reverseFullName2 = {'n', 'a', 'v', 'I', ' ', 'v', 'o', 'n', 'a', 'v', 'I'};
+        char temp2;
+        for (int i = 0; i < reverseFullName2.length /2 ; i++) {
```
Так более гибко, теперь не важно какое будет имя, хоть 100 символов, всё равно будет работать

AleksandrZhukovJava commented on src/Main.java:
```diff
@@ -1,5 +1,78 @@
+        System.out.println(Arrays.toString(reverseFullName2));
+        // Исправлено, как просил Герман
+
+        for (int i = 0; i < reverseFullName.length; i++) {
```
тут скорее имелось ввиду for (int i = reverseFullName.length - 1; i >= 0; i--)

AleksandrZhukovJava commented on src/Main.java:
```diff
@@ -1,5 +1,78 @@
+        System.out.println("задача 2");
+
+        //оставила себе, подумать
+        //double minMonday = monday[0] / 4;
```
Перечитал задание, похоже что деление тут не нужно, мы добавляем в массив траты за неделю, то есть они уже поделены в рамках месяца. Но формулировка конечно не однозначная =)

goldovskaya-m committed:
Нужна помощь=)
Во второй задаче, выводится первое и последнее число цикла. Мах и мин не выводится. Возможно дело в скобках
фигурных?
К 4 задаче коментарий про for(int i = reverseFullName.length - 1; i >= 0; i--) не поняла. И так выводит Иванов Иван,работает?

goldovskaya-m committed:
нашла лишнюю скобку, теперь работает правильно=)!
